src/main/java/net/dries007/tfc/objects/blocks/devices/BlockQuern.java

```diff
@@ -0,0 +1,289 @@
+        }
+    }
+
+    @SuppressWarnings("deprecation")
+    @Override
+    public boolean isOpaqueCube(IBlockState state)
+    {
+        return false;
+    }
+
+    @Override
+    public void breakBlock(World world, BlockPos pos, IBlockState state)
+    {
+        TEQuern teQuern = Helpers.getTE(world, pos, TEQuern.class);
+        if (teQuern != null)
+        {
+            teQuern.onBreakBlock(world, pos);
+        }
+        super.breakBlock(world, pos, state);
+    }
+
+    @Override
+    public boolean onBlockActivated(World world, BlockPos pos, IBlockState state, EntityPlayer playerIn, EnumHand hand, EnumFacing facing, float hitX, float hitY, float hitZ)
+    {
+        if (hand.equals(EnumHand.MAIN_HAND))
+        {
+            ItemStack playerStack = playerIn.getHeldItem(EnumHand.MAIN_HAND);
+
+            TEQuern teQuern = Helpers.getTE(world, pos, TEQuern.class);
+            if (teQuern != null)
+            {
+                boolean isGrinding = teQuern.isGrinding();
+                if (!isGrinding && !playerIn.isSneaking())
+                {
+                    if (playerStack.isEmpty() && facing == EnumFacing.UP && hitX > 0.2f && hitX < 0.4f && hitZ > 0.2f && hitZ < 0.4f && hitY >= 0.875)
+                    {
+                        teQuern.grind();
+                        world.playSound(null, pos, TFCSoundEvents.QUERN_USE, SoundCategory.BLOCKS, 1, 1 + ((world.rand.nextFloat() - world.rand.nextFloat()) / 16));
+                    }
+                    else if (!world.isRemote)
+                    {
+                        TFCGuiHandler.openGui(world, pos, playerIn, TFCGuiHandler.Type.QUERN);
+                    }
+                    return true;
+                }
+                else if (!world.isRemote && playerIn.isSneaking())
+                {
+                    ItemStackHandler inventory = teQuern.getInventory();
+                    ItemStack inputStack = inventory.getStackInSlot(TEQuern.SLOT_INPUT);
+                    ItemStack outputStack = inventory.getStackInSlot(TEQuern.SLOT_OUTPUT);
+
+                    if (hitX > 0.2f && hitX < 0.8f && hitZ > 0.2f && hitZ < 0.8f)
+                    {
+                        if (hitX > 0.4f && hitX < 0.6f && hitZ > 0.4f && hitZ < 0.6f && playerStack.isEmpty() && !inputStack.isEmpty())
+                        {
+                            playerIn.setHeldItem(EnumHand.MAIN_HAND, inventory.extractItem(TEQuern.SLOT_INPUT, inputStack.getCount(), false));
+                            teQuern.setAndUpdateSlots(TEQuern.SLOT_INPUT);
+                            world.playSound(null, pos, SoundEvents.BLOCK_CLOTH_HIT, SoundCategory.BLOCKS, 1, 1);
+                        }
+                        //Can't put an item in there because of "doesSneakBypassUse" is false on the items that can be ground
+                        /*else if (hitX > 0.4f && hitX < 0.6f && hitZ > 0.4f && hitZ < 0.6f && !playerStack.isEmpty() && isValid)
```

Comment:
If this is not needed anymore, remove it. Code comments for things that don't work generally don't add anything documentation wise.